src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c

```diff
@@ -0,0 +1,547 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_ecc_import_export.h"
+#include "pal_utilities.h"
+#include "pal_jni.h"
+#include "pal_eckey.h"
+#include "pal_bignum.h"
+
+int32_t CryptoNative_GetECKeyParameters(
+    const EC_KEY* key,
+    int32_t includePrivate,
+    jobject* qx, int32_t* cbQx,
+    jobject* qy, int32_t* cbQy,
+    jobject* d, int32_t* cbD)
+{
+    assert(qx != NULL);
```

Comment:
What is the logic to assert here as opposed to return an error code? In other functions it seems we do a hard check instead of simply asserting? Perhaps this is an internal function? I would recommend we be consistent here.

Reply:
We guard in the error case for `d` but not here?

Reply:
I guess the asserts are copied from OpenSSL's CryptoNative_GetECKeyParameters

Reply:
The shims are considered 100% as internal functions.  We're inconsistent of asserting vs error checking (largely, IIRC, error checking was first, when we had the possibility of torn state of the native shim package and the caller package... now that we have shared frameworks some of the newer things assert)

Reply:
> The shims are considered 100% as internal functions.
> now that we have shared frameworks some of the newer things assert

W00t. If these functions are internal only I would prefer to limit this sort of validation to asserts only. My assumption is we will always call these functions correctly and find incorrect internal usage with testing.

Reply:
I've noticed that a lot of places in the crypto tests push the validation down to the shim/native library, so we can't use asserts for those ones since they'll take down the process.

Reply:
It all comes back to the current "time-saving" effort of reusing the OpenSSL-based implementations.  The tests (generally) don't care where things fail, just that CryptographicException-derived exceptions happen.  If you make ECDsaAndroid and it does more work in managed code you can do more asserting and less runtime error checking.

If there are tests that care about specifics for the exceptions I believe those should only care about using the concrete public ECDsaOpenSsl type, which should NOT work on Android since it can't speak pointers in an actually interoperable way (and the pointer interop is the only reason the type is public).